src/v/crypto/tests/crypto_bench.cc

```diff
@@ -70,32 +73,48 @@ struct openssl_perf {
     }
 };
 
-PERF_TEST_F(openssl_perf, md5_1k) {
+static std::unique_ptr<openssl_perf> global_perf{nullptr};
```

Comment:
> "This results in some memory being left around and reused even after being
freed which is what was causing the crypto bench test to segfault and fail"

This is a bit hand-wavy. Can you help me understand why it's not a problem during shutdown of Redpanda process?

Reply:
It's a little tough to parse through exactly what is going on inside OpenSSL irt to this issue.  The release build just reports a segfault however in the debug build, I can at least see a "use after free" ASAN error get triggered around the same point where the segfault occurs, so by my first approximation it's the same issue.

I don't believe this is an issue with Redpanda shutdown as the memory is still being free'd (hence the 'use after free') but maybe pointers aren't getting set back to NULL so a subsequent "initialization" doesn't detect that an allocation is required.  As I stated in the commit message, I don't believe OpenSSL is really built to handle init, de-init, and then re-init within the same process, which we are _not_ doing in Redpanda, just in some of the tests.